spec/acceptance/reboot_finished_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper_acceptance'
+
+describe 'Reboot Module - Reboot when Finished' do
+
+  def apply_reboot_manifest(agent, reboot_manifest)
+    apply_manifest_on agent, reboot_manifest do |result|
+      assert_match /defined 'message' as 'step_2'/,
+                   result.stdout, 'Expected step was not finished before reboot'
+    end
+    retry_shutdown_abort(agent)
+  end
+
+  reboot_manifest = <<-MANIFEST
+    notify { 'step_1':
+    } ~>
+    reboot { 'now':
+      apply => finished
+    } ->
+    notify { 'step_2':
+    }
+  MANIFEST
+
+  context 'windows agent' do
+    windows_agents.each do |agent|
+      it 'Reboot After Finishing Complete Catalog' do
```

Comment:
This will create unhelpful output by hiding the agent currently in progress.

You can rewrite the entire thing like this:

```
let(:reboot_manifest) do .... end
agents.each do |agent|
  context "on #{agent}" do
    it 'reboots after finishing the complete catalog' do
      apply_manifest_on agent, reboot_manifest do |result|
        assert_match /defined 'message' as 'step_2'/,
                     result.stdout, 'Expected step was not finished before reboot'
      end
      retry_shutdown_abort(agent)
    end
  end
end
```

Since the agents don't seem to differ in their test setup and execution, this should be one block for both lists. That also removes the need for the (unconventional) helper method, and reduces duplication.